build_tools/cmake/build_and_test_asan.sh

```diff
@@ -22,7 +22,6 @@ BUILD_DIR="${1:-${IREE_ASAN_BUILD_DIR:-build-asan}}"
 IREE_ENABLE_ASSERTIONS="${IREE_ENABLE_ASSERTIONS:-ON}"
 
 source build_tools/cmake/setup_build.sh
-source build_tools/cmake/setup_ccache.sh
```

Comment:
Ah, I was going to continue using this script or fork it to `setup_sccache.sh`. These scripts are designed for _both_ CI and developer use, and local developer use benefits from flexibility in choosing where to build and which cache settings to use.

I'm also expecting that as we figure out how to get sccache read-only access using Azure for storage, we'll want to put that logic in a script that can be reused in each build script, instead of fully inlining the cache calls here.

Reply:
fixed

Reply:
We shouldn't configure both ccache and sccache. See the commit I shared the other day: https://github.com/ScottTodd/iree/commit/25a5ffb1fc25ff137a94bbb68313803dcdc01325. That adds a branch in `build_tools/cmake/setup_ccache.sh` that configures for Azure instead of GCP. The problem is different now that we're using sccache for Azure and ccache for GCP / local though.

Maybe something like `IREE_READ_REMOTE_AZURE_SCCACHE` would set `IREE_USE_SCCACHE` and then the scripts here would do
```bash
if (( IREE_USE_CCACHE == 1 )); then
  ccache --show-stats
fi

if (( IREE_USE_SCCACHE == 1 )); then
  sccache --show-stats
fi
```

That sounds complicated though, so how about instead creating a new `setup_sccache.sh` script and having these build scripts just use that? The main thing to look out for is that the cache should be optional, and the scripts should be runnable locally.

Reply:
Can you review https://github.com/iree-org/iree/pull/18499 then try switching these scripts from using `source build_tools/cmake/setup_ccache.sh` to use the new `source build_tools/cmake/setup_sccache.sh`?

Reply:
Oh, you pushed right when I sent that comment. Reviewing now.